src/web-auth/redirect.js

```diff
@@ -1,74 +1,33 @@
-var UsernamePassword = require('./username-password');
-var objectHelper = require('../helper/object');
+var CrossOriginAuthentication = require('./cross-origin-authentication');
 var Warn = require('../helper/warn');
-var assert = require('../helper/assert');
 
-function Redirect(client, options) {
+function Redirect(auth0, options) {
   this.baseOptions = options;
-  this.client = client;
+  this.client = auth0.client;
+  this.crossOriginAuthentication = new CrossOriginAuthentication(auth0, this.baseOptions);
 
   this.warn = new Warn({
     disableWarnings: !!options._disableDeprecationWarnings
   });
 }
 
 /**
- * @callback credentialsCallback
- * @param {Error} [err] error returned by Auth0 with the reason of the Auth failure
- * @param {Object} [result] result of the AuthN request
- * @param {String} result.accessToken token that can be used with {@link userinfo}
- * @param {String} [result.idToken] token that identifies the user
- * @param {String} [result.refreshToken] token that can be used to get new access tokens from Auth0. Note that not all clients can request them or the resource server might not allow them.
- */
-
-/**
- * Performs authentication with username/email and password with a database connection
- *
- * This method is not compatible with API Auth so if you need to fetch API tokens with audience
- * you should use {@link authorize} or {@link login}.
+ * Logs in the user with username and password using the cross origin authentication (/co/authenticate) flow. You can use either `username` or `email` to identify the user, but `username` will take precedence over `email`.
+ * This only works when 3rd party cookies are enabled in the browser. After the /co/authenticate call, you'll have to use the {@link parseHash} function at the `redirectUri` specified in the constructor.
  *
```

Comment:
It's not entirely true that it works only when 3rd party cookies are enabled, depends on the browser. Maybe we can link to the Cross Origin docs

Reply:
updated